src/main/java/com/exadel/discount/util/AccessTokenUtil.java

```diff
@@ -0,0 +1,63 @@
+package com.exadel.discount.util;
+
+import com.exadel.discount.model.token.TokenPayload;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class AccessTokenUtil {
+
+    private String SECRET_KEY = "accessSecret";
+
+    public String extractUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public Date extractExpiration(String token) {
+        return extractClaim(token, Claims::getExpiration);
+    }
+
+    public <T> T extractClaim(String token, Function<Claims, T> claimsResolver) {
+        final Claims claims = extractAllClaims(token);
+        return claimsResolver.apply(claims);
+    }
+
+    private Claims extractAllClaims(String token) {
+        return Jwts.parser().setSigningKey(SECRET_KEY).parseClaimsJws(token).getBody();
+    }
+
+    private Boolean isTokenExpired(String token) {
+        return extractExpiration(token).before(new Date());
+    }
+
+    public String generateToken(TokenPayload payload) {
+        Map<String, Object> claims = new HashMap<>();
+        Map<String, Object> roleClaim = new HashMap<>();
+        roleClaim.put("role", payload.getRole());
+
+        return createToken(claims, roleClaim, payload.getEmail());
+    }
+
+    private String createToken(Map<String, Object> claims, Map<String, Object> roleClaim, String subject) {
+        return Jwts.builder()
+                .setClaims(claims)
+                .addClaims(roleClaim)
+                .setSubject(subject)
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + 1000 * 60 * 60)) // 1 hour
```

Comment:
expiration should be configurable and come from the environment variables

Reply:
And return LocalDateTime type, like around this project...?

Reply:
This is done.